tests/models/encoder_decoders/wikitables_semantic_parser_test.py

```diff
@@ -25,6 +24,82 @@ def setUp(self):
     def test_model_can_train_save_and_load(self):
         self.ensure_model_can_train_save_and_load(self.param_file)
 
+    def test_correct_neighbor_indexes(self):
+        # Verify that 2nd entity has correct neighbor indexes.
+        worlds = []
+        for instance in self.dataset.instances:
+            worlds.append(instance.fields['world'].metadata)
+        num_entities = len(worlds[0].table_graph.entities)
+        tensor = Variable(torch.LongTensor([]))
+        neighbor_indexes = self.model._get_neighbor_indexes(worlds, num_entities, tensor)
+        assert neighbor_indexes.size(1) == num_entities
+        assert neighbor_indexes.size(2) == num_entities
+        lst = [0]*(num_entities)
+        lst[0], lst[1] = 45, 45
+        second_entity_neighbors = (torch.LongTensor(lst))
+        assert torch.equal(neighbor_indexes[0][1].data, second_entity_neighbors)
+
+    def test_all_types_present_in_type_vector(self):
+        # Verifies that at least 1 instance of every one hot vector
```

Comment:
Again here I think it'd be more informative to just construct a small world, and make sure that you get the results you expect for both return values from `_get_type_vector`.  If the world has both entity types, checking for the explicit values will also implicitly check that all types are present in the returned vector.